### PR TITLE
Bring the natural sort out of the utils package.

### PR DESCRIPTION
**Pull request by howbazaar**
Extracts the natural sorting from the bloated juju/utils package.

### CODE REVIEW

File: naturalsort_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+func copyStrSlice(in []string) []string {
+	out := make([]string, len(in))
+	copy(out, in)
+	return out
+}
+
+func shuffle(a []string) {
+	// See https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#Modern_method
```
Comment (jameinel) [inline naturalsort_test.go:139]: I believe there are actually math/rand Perm and Shuffle already

File: naturalsort.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	"unicode"
+)
+
+// Sort sorts strings according to their natural sort order.
+func Sort(s []string) []string {
+	sort.Sort(naturally(s))
+	return s
+}
+
+type naturally []string
```
Comment (jameinel) [inline naturalsort.go:19]: Probably worth exposing this type. That way you could
sort.Sort(naturalsort.Strings(slice))
or
sort.StableSort(naturalsort.Strings(slice))

etc.
I'm happy to leave the helper Sort  and I'm open to suggestion on the name of the type.
Strings is probably not ideal as sort.Strings is a thing that has specific meaning (sort a slice of strings).

**Review by jameinel — APPROVED** [review]
I really like pulling this into its own package.A couple of comments on the names but overall much thanks for doing this.

File: naturalsort_test.go (hunk 1 above)
Comment (howbazaar) [reply naturalsort_test.go:139]: There is, but we'd still need the shuffle method for our testing. We'd also need to change the implementation to fit into rand.Shuffle, and I don't think it is worth the effort here.

File: naturalsort.go (hunk 2 above)
Comment (howbazaar) [reply naturalsort.go:19]: I'm not convinced. Partly I'm against providing multiple ways to do the same thing, and also when dealing just with strings, the concept of a stable sort is a bit wasted.

**Pull request merged**